### ISSUE TITLE
Switching Extension and Forwarder implementations not working

### ISSUE THREAD

**Issue by VladimirRubin** [DataDog/serverless-plugin-datadog#142]
## Expected Behavior
We should be easily able to switch/change "serverless-datadog-plugin" settings between "forwarder" and "addExtension" implementation

## Actual Behavior
Switching "forwarder" -> "addExtension" -> "forwarder" settings produce an error (see "Stacktrace" below)

## Steps to Reproduce the Problem

Minimal Reproducible Example: https://github.com/VladimirRubin/serverless-datadog-plugin-test

  1. `serverless deploy --datadog-api-key <apiKey> --datadog-type forwarder --datadog-forwarder <forwarderArn>`  -> deploy is passing
  1. `serverless deploy --datadog-api-key <apiKey> --datadog-type extension` -> deploy is passing
  1. `serverless deploy --datadog-api-key <apiKey> --datadog-type forwarder --datadog-forwarder <forwarderArn>` -> deploy failing with the error (see "Stacktrace" below)

## Specifications

  - Serverless Framework version: 1.83.2
  - Datadog Serverless Plugin version: 2.27.0
  - Lambda function runtime (Python 3.7, Node 10, etc.): Node 14
  
## Stacktrace
  
  ```
An error occurred: ExecutionLogGroup - Resource handler returned message: "Resource of type 'AWS::Logs::LogGroup'
with identifier '{"/properties/LogGroupName":"API-Gateway-Execution-Logs_<restApiID>/development"}' already exists."
(RequestToken: <requestToken>, HandlerErrorCode: AlreadyExists).
  ```

As a workaround setup `subscribeToApiGatewayLogs: false` helps

**Comment by jcstorms1** [comment]
Hey Vladimir thanks for reaching out and opening this up. Unfortunately, I've been unable to reproduce this behavior using the test setup you provided. I attempted with Serverless framework version 1.83.2 and 2.51, but I was not able to trigger the error you shared above. 

I checked the code, but I did not find anything that would trigger an error like this when enabling and disabling the extension. However, I did some additional searching and found similar references to the error coming from Serverless. Based on what I found, I think the issue might be that Cloudformation mishandled the existing log group during the deploy somehow. I'd be interested to know if you have tried deleting the log group and running through the steps again to see if it still occurs?

**Comment by tianchu** [comment]
@VladimirRubin Do you still experience this issue? The error seems to be caused by https://github.com/DataDog/serverless-plugin-datadog/blob/f6c92cb6b007e8b520210075bed14783e837dc11/src/forwarder.ts#L206, however, CFN is not expected to recreate a resource that is already defined in the stack and created before.

**Comment by VladimirRubin** [comment]
> @VladimirRubin Do you still experience this issue? The error seems to be caused by
> 
> https://github.com/DataDog/serverless-plugin-datadog/blob/f6c92cb6b007e8b520210075bed14783e837dc11/src/forwarder.ts#L206
> 
> , however, CFN is not expected to recreate a resource that is already defined in the stack and created before.

Yes, unfortunately, the problem is still reproducible on the example repo I attached to the ticket.

**Comment by tianchu** [comment]
Thanks for getting back to us @VladimirRubin, we will try to reproduce this again using your testing project, there might be something different in our set up and yours.

**Comment by tianchu** [comment]
OK, once I added the these lines to your sample project I can reproduce the issue.

```yaml
provider:
  name: aws
  runtime: nodejs14.x
  stage: development
  region: eu-central-1
  # needed to add the following two lines
  logs:
    restApi: true
```

Now it starts to make sense, because without these two lines, `subscribeToApiGatewayLogs` will not do anything.

@jcstorms1 perhaps you could take another deeper look now that we know how to reproduce it? I also noticed two interesting things:
- `subscribeToApiGatewayLogs` doesn't do anything when using extension instead of the forwarder, which is unexpected. My guess is that you added the logic to subscribe to api gateway log groups to the same place we subscribe the forwarder to lambda log groups?
- After deleting the ExecutionLogGroup resource in CloudFormation, the actual log group doesn't seem to be deleted in CloudWatch.

**Comment by jcstorms1** [comment]
Hey @VladimirRubin I wanted to let you know we just released [v3.0.1 ](https://github.com/DataDog/serverless-plugin-datadog/tree/v3.0.1) of the plugin and I believe it should resolve the issues you were having here. The issue was related to Execution log groups not being removed by CloudFormation when stacks are deleted. When redeploying via the plugin we were trying to add the Execution log group but it was already created.

We've now moved this option into its own parameter, `subscribToExecutionLogs`, and it is disabled by default. Moving forward, if you need to subscribe to these log groups and want to delete them at a later point, it will require manual deletion since deleting the stack will not remove them. Please let us know if continue to see any issues after trying the above changes.
